facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -357,4 +397,19 @@ public TimeoutConfiguration getTimeoutConfiguration() {
     public void setTimeoutConfiguration(TimeoutConfiguration timeoutConfiguration) {
         this.timeoutConfiguration = timeoutConfiguration;
     }
+
+    /**
+     * Shutdown thread pool executor for all facades.
+     *
+     */
+    public static void shutdown() {
```

Comment:
Guess in a quick glance it looks like annotation @PreDestroy would be
needed for that to happen.  I don't see how declaring a static method on a
pojo will result in the EJB calling that method on shutdown.

On Wed, May 18, 2016 at 10:05 AM, Marek <notifications@github.com> wrote:

> In
> facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350#discussion_r63708374>
> :
>
> > @@ -357,4 +397,19 @@ public TimeoutConfiguration getTimeoutConfiguration() {
> >      public void setTimeoutConfiguration(TimeoutConfiguration timeoutConfiguration) {
> >          this.timeoutConfiguration = timeoutConfiguration;
> >      }
> > +
> > +    /**
> > +     * Shutdown thread pool executor for all facades.
> > +     *
> > +     */
> > +    public static void shutdown() {
>
> Not sure what you mean. According to that thread, FacadeService.shutdown()
> could be called on singleton bean's pre destroy to cleanup the pool on
> undeployment.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350/files/17b676bee35728c0487e52415290b9e02d5b9cc0#r63708374>
>